stage1/init/init.go

```diff
@@ -230,23 +228,15 @@ func getArgsEnv(p *Pod, flavor string, systemdStage1Version string, debug bool)
 
 	switch flavor {
 	case "coreos":
-		// when running the coreos-derived stage1 with unpatched systemd-nspawn we need some ld-linux hackery
-		args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), interpBin))
 		args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), nspawnBin))
 		args = append(args, "--boot") // Launch systemd in the pod
 
-		// Note: the coreos flavor uses systemd-nspawn v215 but machinedRegister()
-		// checks for the nspawn registration method used since v216. So we will
-		// not register when the host has systemd v215.
 		if machinedRegister() {
```

Comment:
What I understand is that machinedRegister only checks nspawn v216+ so before it was failing always and now it is necessary. This comment only explains that the function will fail for v215 and I removed it because we don't have v215 in stage1 now. We can probably also remove the comment mentioning v215 in the function itself and the TODO there.

@alban?